lms/djangoapps/courseware/access.py

```diff
@@ -174,7 +187,14 @@ def has_staff_access_to_preview_mode(user, course_key):
     """
     has_admin_access_to_course = any(administrative_accesses_to_course_for_user(user, course_key))
 
-    return has_admin_access_to_course or is_masquerading_as_student(user, course_key)
```

Comment:
The original test was role-based, while here we have a permissions-based test. Does it continue to work as intended if permissions assignments change? As I suggested in a different comment, I believe a requirement for the roles-permissions app is that permissions granted at runtime to one role should never be affected by configuration changes to permissions made to another role. If a permissions-based test is most appropriate here, then we might be missing a permission: PREVIEW_ALL_CONTENT.